### DISCUSSION TITLE
Onboarding Q&A

### DISCUSSION THREAD

**Discussion by dougkeiller** [nicobrauchtgit/obsidian-Smart2Brain#87] [Q&A]
What is the "Documents to Retrieve" setting? How should I think about the best setting for my use case / vault? Thanks!

**Comment by dougkeiller** [comment]
What does the time to index mean? How can I estimate OpenAI costs? How often will this happen?

![image](https://github.com/your-papa/obsidian-Smart2Brain/assets/128188043/ab29e284-e9e8-41dd-b23a-a7a728ab234e)

**Reply by Leo310** [reply]
> What does the time to index mean?

When initializing the S2B your whole Vault needs to be indexed once meaning that your notes are transformed into embedding vectors with which we can generate personalized responses. This process takes some time depending on your computing power.

> How can I estimate OpenAI costs? 

Difficult to determine atm but should not be much. We could show an estimation before indexing in the future.

> How often will this happen?

it only indexes your whole vault once at plugin startup.
After the initial indexing, subsequent indexing cycles should be way faster as notes that are indexed once won't be indexed again except if their content has changed.

**Comment by Leo310** [comment]
> What is the "Documents to Retrieve" setting? How should I think about the best setting for my use case / vault? Thanks!

This setting specifies the maximum amount of documents that are retrieved from your vault and passed to the LLM. The higher the number the bigger the context.

While we’re working on gathering best practices, I recommend experimenting to find what works best for your specific use case.
Keep in mind that the plugin is still in its early stages, and we’re actively exploring ways to optimize its usage.

**Comment by j2l** [comment]
May I ask a question related to onboarding here?

Before installing S2B, I understand that it indexes my opened vault ... which is not what I want right now (975 files, various subjects).
Could I select another vault (folder) to be indexed?
Could I have 2 vaults opened at the same time? One indexed for S2B and another one not indexed?
Thank you!

**Reply by nicobrauchtgit** [reply]
Hi @j2l, the two vaults are independent form each other and you can absolutely have a vault that is indexed and one that is not. You don't even have to install the plugin in the other vault.

**Reply by j2l** [reply]
OK! Thank you very much @nicobrauchtgit

**Comment by jychen6688** [comment]
when support Chinese

**Comment by jluedke** [comment]
I run ollama on a server i control, and connect via ssh tunnel.  In the addon configuration I can see my models listed in the dropdown window and select it.

When I launch the addon the Setup side pane gives me install instructions. When i click the 'Test' button I just see a red X appear.  I have verified that my ollama instance is running correctly, and I can connect to it and query it on the configured local port.  Do I need to have ollama installed on the local machine in order to run this addon? Will it work with only the network API?

**Reply by jluedke** [reply]
I have  debugged the issue a little further, i'm getting a 403.

Request
```
GET /api/tags HTTP/1.1
Accept: */*
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US
Connection: keep-alive
Host: 127.0.0.1:11434
Origin: app://obsidian.md
Sec-Fetch-Mode: cors
Sec-Fetch-Site: cross-site
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) obsidian/1.6.3 Chrome/120.0.6099.291 Electron/28.3.3 Safari/537.36
sec-ch-ua-mobile: ?0
sec-ch-ua-platform: "Linux"
```

Response
```
HTTP/1.1 403 Forbidden
Content-Length: 0
Date: Sun, 23 Jun 2024 14:26:26 GMT
```

Anyhow, I solved the issue by updating the env variable OLLAMA_ORIGINS=* in my Docker image

**Comment by Leo310** [comment]
Closing this as for the next major Release we will create a new fresh discussion on onboarding.
Wont take long anymore!
